src/main/java/net/atos/entng/support/filters/OwnerOrLocalAdmin.java

```diff
@@ -19,85 +19,100 @@
-		if (functions != null  && (functions.containsKey(DefaultFunctions.ADMIN_LOCAL) || functions.containsKey(DefaultFunctions.SUPER_ADMIN))) {
-			// If current user is a local admin, check that its scope contains the ticket's school_id
-			Function adminLocal = functions.get(DefaultFunctions.ADMIN_LOCAL);
-            // super_admin always are authorized
-			if (adminLocal != null && adminLocal.getScope() != null && !adminLocal.getScope().isEmpty()) {
-				query.append(" OR t.school_id IN (");
-				for (String scope : adminLocal.getScope()) {
-					query.append("?,");
-					values.add(scope);
+			if (ticketIds.isEmpty()) {
+				handler.handle(false);
+				return;
+			}
+			JsonArray values = new JsonArray();
+			StringBuilder query = new StringBuilder("SELECT count(*) FROM support.tickets AS t");
+			query.append(" WHERE t.id IN (");
+			bindInSqlKeyword(query, values, new HashSet<>(ticketIds));
+			query.append(")");
+			query.append(" AND (t.owner = ?"); // Check if current user is the ticket's owner
+			values.add(user.getUserId());
+
+			Map<String, UserInfos.Function> userFunctions = user.getFunctions();
+			if (userFunctions != null  && (userFunctions.containsKey(DefaultFunctions.ADMIN_LOCAL) || userFunctions.containsKey(DefaultFunctions.SUPER_ADMIN))) {
+				// If current user is a local admin, check that its scope contains the ticket's school_id
+				Function adminLocal = userFunctions.get(DefaultFunctions.ADMIN_LOCAL);
+				// super_admin always are authorized
+				if (adminLocal != null && adminLocal.getScope() != null && !adminLocal.getScope().isEmpty()) {
+					query.append(" OR t.school_id IN (");
+					bindInSqlKeyword(query, values, new HashSet<>(adminLocal.getScope()));
+					query.append(")");
 				}
-				query.deleteCharAt(query.length() - 1);
-				query.append(")");
 			}
-		}
-
-		query.append(")");
-        admin = functions.get(DefaultFunctions.SUPER_ADMIN);
-
-
-        final Function finalAdmin = admin;
-        Sql.getInstance().prepared(query.toString(), values, new Handler<Message<JsonObject>>() {
-			@Override
-			public void handle(Message<JsonObject> message) {
+			query.append(")");
+			Sql.getInstance().prepared(query.toString(), values, result -> {
 				request.resume();
-				Long count = SqlResult.countResult(message);
-
-                if( finalAdmin != null  ) {
-                    handler.handle(true);
-                } else {
-                    handler.handle(count != null && count > 0);
-                }
-			}
+				Long count = SqlResult.countResult(result);
+				if(userFunctions.get(DefaultFunctions.SUPER_ADMIN) != null) {
+					// super admin is authorized
+					handler.handle(true);
+				} else {
+					handler.handle(count != null && count == ticketIds.size());
+				}
+			});
 		});
+	}
 
+	private Future<Set<Integer>> getTicketIdsFromBody(HttpServerRequest request) {
+		Promise<Set<Integer>> promise = Promise.promise();
+		HashSet<Integer> idsFromBody = new HashSet<>();
```

Comment:
pour respecter le principe de "Coding to interfaces, not implementation" il serait mieux d'avoir cette déclaration-là pour `idsFromBody` :

```
final Set<Integer> idsFromBody = new HashSet<>();
```